### PR TITLE
Adding the ability to repeat payloads for the sessions workload

### PR DESCRIPTION
**Pull request by cedrichansen**
- Payloads can be repeated by using the `repetitionDelay` keyword in xml configuration files
- View change in README for documentation on the change
- view config/config_repetitionDelayExample.xml for an example configuration file

Signed-off-by: Cedric Hansen <cedric.hansen@ibm.com>

### CODE REVIEW

**Comment by cedrichansen** [comment]
@smlambert Is there a good way to take the changes I made on my copy of SynthetiGCWorkload and commit them into this repo? I made far too many changes in my untraced version of the project, to be able to pick out what I did and insert it into each class in this repo... It's very hard to follow what I added when looking at the files changed

**Comment by cedrichansen** [comment]
Tested the changes locally with the SyntheticGCWorkload jar and the example configuration files (config_repetitionDelayExample.xml), and the behaviour is the same as previously observed in my standalone copy of SGCW (ie, is working as expected). @smlambert These changes are ready for review

File: functional/SyntheticGCWorkload/config/config_repetitionDelayExample.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<!-- example to use a repetion delay to repeat a payload
```
Comment (smlambert) [inline functional/SyntheticGCWorkload/config/config_repetitionDelayExample.xml]: spelling repetion -> repetition

File: functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/Workloads.java [hunk 2]
```diff
@@ -370,7 +375,7 @@ public String toString()
 	/**
 	 * A thread that displays status updates to the user. 
 	 *
-	 *  
+	 * @author Andrew Somerville <andrew.somerville@unb.ca>
```
Comment (smlambert) [inline functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/Workloads.java]: Please remove author tags in source code, we have kept Andrew's name in the docs, but as we open-source code, we strip out author tags in src.

File: functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/configuration/PayloadSetConfiguration.java [hunk 3]
```diff
@@ -28,398 +28,605 @@
...
+	public int getNumberOfRepeatedPayloads() {
+
+		int numberOfRepeatedPayloads = 0;
+
+		if (this.childPayloadSetConfigurations.length == 0) {
+			// we are a leaf, add payloads
+			int numberOfRepetitions = (int) (this.shutdownTime / this.repetitionDelay);
+			if (numberOfRepetitions == 0) {
+				/*
+				 * Due to max duraration and rep delay configurations, we need to *at least*
```
Comment (smlambert) [inline functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/configuration/PayloadSetConfiguration.java]: spelling duraration -> duration

File: functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/configuration/WorkloadConfiguration.java [hunk 4]
```diff
@@ -28,214 +28,312 @@
...
-		// Ensure we've consumed all symbols. 
-		getElement().checkForUnusedAttributes();	
+		// Ensure we've consumed all symbols.
+		getElement().checkForUnusedAttributes();
 	}
-	
+
 	/**
-	 * Gets the ID of the parsed element. 
+	 * Get the repition delay for workload
```
Comment (smlambert) [inline functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/configuration/WorkloadConfiguration.java]: repetition

File: functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/configuration/WorkloadConfiguration.java [hunk 5]
```diff
@@ -28,214 +28,312 @@
...
+				PayloadSetConfiguration[] repeatedPayloads = null;
+				try {
+					repeatedPayloads = updatePayload(psc);
+				} catch (CloneNotSupportedException e) {
+					// This should never happen
+					System.out.println("Clone not supported... Remove repetition delay");
+					e.printStackTrace();
+				}
+
+				/** -1 necessary because we do not double count the original payloadSetConfiguration. Doing -1 effectively ommits the doubled up payload */
```
Comment (smlambert) [inline functional/SyntheticGCWorkload/src/net/adoptopenjdk/casa/workload_sessions/configuration/WorkloadConfiguration.java]: spelling ommits -> omits
